src/esp/scene/SceneNode.h

```diff
@@ -41,6 +58,7 @@ class SceneNode : public MagnumObject,
   // terminate node (e.g., "MagnumScene" defined in SceneGraph) as its ancestor
   SceneNode() = delete;
   SceneNode(SceneNode& parent);
+  ~SceneNode() override;
```

Comment:
By the way, I know in the base class, there is a function called `setParent(parentNode)` which moves a subtree rooted at "this" node to another node `parentNode`.

Certainly you need to overload that function as well. In this function:
- You need to check if the parentNode is a leaf node.
- Update the sensorSuites in the old ancestors
- setParent
- Update the sensorSuites in the new ancestors

This is quite challenging. Please think very carefully:
- when "this" node is a `leaf` node.
- when "this" node is NOT a `leaf` node.

Reply:
Be CAREFUL! No duplicate code.
Expect some code will be used in both ctor and setParent, also dctor and setParent. 
You need to extract a function to accommodate such code.